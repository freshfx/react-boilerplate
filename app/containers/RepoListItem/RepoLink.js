import styled from '@emotion/styled'
import NormalA from 'components/Anchor'

const RepoLink = styled(NormalA)`
  height: 100%;
  color: black;
  display: flex;
  align-items: center;
  width: 100%;
`

export default RepoLink
