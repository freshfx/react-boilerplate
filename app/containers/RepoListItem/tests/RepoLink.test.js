import React from 'react'
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import RepoLink from '../RepoLink'

describe('<RepoLink />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(<RepoLink />).toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })

  it('should have a className attribute', () => {
    const renderedComponent = shallow(<RepoLink />)
    expect(renderedComponent.prop('className')).toBeDefined()
  })

  it('should adopt a valid attribute', () => {
    const id = 'test'
    const renderedComponent = shallow(<RepoLink id={id} />)
    expect(renderedComponent.prop('id')).toEqual(id)
  })
})
