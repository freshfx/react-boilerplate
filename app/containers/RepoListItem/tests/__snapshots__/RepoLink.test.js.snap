// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoLink /> should match snapshot 1`] = `
.c0 {
  color: #41addd;
  height: 100%;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c0:hover {
  color: #6cc0e5;
}

<RepoLink>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "RepoLink-pvpwpn-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "color:#41addd;&:hover{color:#6cc0e5;}",
            "height:100%;color:black;display:flex;align-items:center;width:100%;",
          ],
        },
        "displayName": "RepoLink",
        "foldedComponentIds": Array [
          "Anchor-aulrtc-0",
        ],
        "render": [Function],
        "styledComponentId": "RepoLink-pvpwpn-0",
        "target": "a",
        "toString": [Function],
        "usesTheme": false,
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <a
      className="c0"
    />
  </StyledComponent>
</RepoLink>
`;
